### PR TITLE
SONAR-6855: Increase Tomcat maxHttpHeaderSize to 48 kb in version 5.2

### PR DESCRIPTION
**Pull request by sulabh-msft**
SONAR-6855: Increase Tomcat's maxHttpHeaderSize to 48 kb in version 5.2

The default value of maxHttpHeaderSize is increased to 48 kb in SonarQube 5.2 . With this change both http and https connector will support header size up to 48 kb. SonarQube server needs to support 48 kb header size to accommodate the authentication token used for negotiate protocol of windows authentication.

Testing
1. Testing of the sonarqube server's http/https connector with the changes.
2. Added unit test for the changes

### CODE REVIEW

File: server/sonar-web/src/main/webapp/WEB-INF/lib/need_authentication.rb [hunk 1]
```diff
@@ -76,7 +76,7 @@ def external_auth(username, password, servlet_request, details)
     if @java_authenticator
```
Comment (sulabh-msft) [inline server/sonar-web/src/main/webapp/WEB-INF/lib/need_authentication.rb]: Please ignore the changes in this files. These changes are done as part of the SONAR-6856 ( We have created separate pull request for this)

SONAR-6856: Add a new API ExternalGroupsProvider.doGetGroups(Context context) to allow plugins to pass groups information to SonarQube

File: sonar-plugin-api/src/main/java/org/sonar/api/security/ExternalGroupsProvider.java [hunk 2]
```diff
@@ -19,20 +19,54 @@
  */
 package org.sonar.api.security;
```
Comment (sulabh-msft) [inline sonar-plugin-api/src/main/java/org/sonar/api/security/ExternalGroupsProvider.java]: Please ignore the changes in these files. These changes are done as part of the SONAR-6856 ( We have created separate pull request for this)

SONAR-6856: Add a new API ExternalGroupsProvider.doGetGroups(Context context) to allow plugins to pass groups information to SonarQube

File: sonar-plugin-api/src/test/java/org/sonar/api/security/ExternalGroupsProviderTest.java [hunk 3]
```diff
@@ -0,0 +1,109 @@
+/*
```
Comment (sulabh-msft) [inline sonar-plugin-api/src/test/java/org/sonar/api/security/ExternalGroupsProviderTest.java]: Please ignore the changes in these files. These changes are done as part of the SONAR-6856 ( We have created separate pull request for this)

SONAR-6856: Add a new API ExternalGroupsProvider.doGetGroups(Context context) to allow plugins to pass groups information to SonarQube

**Comment by simonbrandhof** [comment]
Hi,
As discussed the size does not need to be configurable. It should be hardcoded to 48kb.
Thanks

**Comment by sulabh-msft** [comment]
maxHttpHeaderSize  value is now hard coded to 48 kb. Now it is not configurable.

**Comment by simonbrandhof** [comment]
Changes applied but this PR was not merged. Some commits are out of scope.

**Comment by sulabh-msft** [comment]
Thanks! Simon

**Pull request closed without merging**